### PR TITLE
Optionally display root port pci device id

### PR DESCRIPTION
**Pull request by 0xff43**
Added an option to pcm-iio to display root port PCIe device ID
to help identify the source of IO. This change is useful for
in automation scripts.

### CODE REVIEW

**Comment by opcm** [comment]
thanks for the patch! I will be back next week and review it properly

**Comment by opcm** [comment]
could you please post a sample pcm-iio output with this new option here?

**Comment by 0xff43** [comment]
Please see attached files
[without_rootport.log](https://github.com/opcm/pcm/files/8962358/without_rootport.log)
[with_rootport.log](https://github.com/opcm/pcm/files/8962359/with_rootport.log)
.

File: src/pcm-iio.cpp [hunk 1]
```diff
@@ -346,12 +346,37 @@ std::string build_csv_row(const std::vector<std::string>& chunks, const std::str
...
+    std::string rp_pci;
+
+    if (!show_root_port)
+        return rp_pci;
+
+    for (auto part = stack->parts.begin(); part != stack->parts.end(); part = std::next(part))
+    {
+        if (part->part_id == part_id)
+        {
+            std::snprintf(tmp, sizeof(tmp), "%02x:%02x.%02x", part->root_pci_dev.bdf.busno,
```
Comment (0xff43) [inline src/pcm-iio.cpp:361]: Function should not be %02x, it should just be %x. I will fix it.

Comment (opcm) [reply src/pcm-iio.cpp:361]: ok, waiting for the update

Comment (0xff43) [reply src/pcm-iio.cpp]: Done

**Comment by opcm** [comment]
interesting, github shows now that the changes have a conflict and can't be merged

**Review by opcm — APPROVED** [review]

**Comment by opcm** [comment]
seems like a github glitch. it is merged: https://github.com/opcm/pcm/commits/master

**Pull request closed without merging**